### PR TITLE
Allow Symfony 6

### PR DESCRIPTION
**Pull request by franmomu**
supersedes https://github.com/FriendsOfSymfony/FOSElasticaBundle/pull/1840

This PR adds a CI job to test with Symfony 6, ~the `ConfigSourcePassTest` has been changed because there was a problem with `replaceArgument` not returning anything and I decided to change it to use an actual `ContainerBuilder`.~ (solved in https://github.com/FriendsOfSymfony/FOSElasticaBundle/pull/1845)

### CODE REVIEW

File: tests/Functional/WebTestCase.php [hunk 1]
```diff
@@ -70,4 +72,14 @@ protected static function getVarDir()
     {
         return \substr(\strrchr(static::class, '\\'), 1);
     }
+
+    // To be removed when dropping support of Symfony < 5.3 and use "self::getContainer()" instead.
```
Comment (XWB) [inline tests/Functional/WebTestCase.php:76]: Feel free to bump to Symfony 5.4 and remove this BC layer.

Comment (franmomu) [reply tests/Functional/WebTestCase.php:76]: You mean dropping support of Symfony 4.4?

Comment (XWB) [reply tests/Functional/WebTestCase.php]: Oops, I forgot we still support 4.4 :)

**Comment by mamchyts** [comment]
any updates?

**Comment by franmomu** [comment]
Now the Symfony 6 job runs on PHP 8.1, I think this ir ready for review again

**Comment by mamchyts** [comment]
@franmomu  are you sure about migration from 8.0 to 8.1?

https://github.com/symfony/symfony/blob/6.0/composer.json:
```
    "require": {
        "php": ">=8.0.2",
```

**Review by XWB — APPROVED** [review]

**Comment by franmomu** [comment]
> @franmomu are you sure about migration from 8.0 to 8.1?
> 
> https://github.com/symfony/symfony/blob/6.0/composer.json:
> 
> ```
>     "require": {
>         "php": ">=8.0.2",
> ```

My message was vague, what I meant is that we weren't testing anything with PHP `8.1` in the CI, not that Symfony 6 requires `8.1`.

I'll add more jobs to the matrix

**Comment by XWB** [comment]
Thank you @franmomu

**Pull request merged**